tests/main/snapd-state/task.yaml

```diff
@@ -14,9 +14,21 @@ execute: |
     "$TESTSTOOLS"/snapd-state --foo 2>&1 | MATCH "snapd-state: no such command: --foo"
     "$TESTSTOOLS"/snapd-state foo 2>&1 | MATCH "snapd-state: no such command: foo"
 
+    function check_date() {
+        local TEST_DATE="$1"
+        local TOLERANCE_SECONDS=1800 # 30 minutes
+
+        local TEST_TIMESTAMP=$(date +'%s' --date="${TEST_DATE}")
```

Comment:
Could please add a comment explaining the problem with this check around midnight, explaining why it's 30 minutes? Do we actually need such high tolerance if we got rid of MATCH?

Reply:
Done!